pkg/datapath/iptables/iptables.go

```diff
@@ -1217,7 +1219,77 @@ func (m *IptablesManager) installMasqueradeRules(prog iptablesInterface, ifName,
 		}
 	}
 
-	// Masquerade all egress traffic leaving the node
+	// Masquerade egress traffic leaving the node based on source routing
+	//
+	// If this option is enabled, then it takes precedence over the catch-all
+	// MASQUERADE further below.
+	if option.Config.EnableMasqueradeRouteSource {
+		devices := option.Config.GetDevices()
+		if option.Config.EgressMasqueradeInterfaces != "" {
+			devices = strings.Split(option.Config.EgressMasqueradeInterfaces, ",")
+		}
+		family := netlink.FAMILY_V4
+		if prog == ip6tables {
+			family = netlink.FAMILY_V6
+		}
+		if option.Config.EnableIPv4 && prog == ip4tables ||
```

Comment:
I think we don't need to check for `EnableIPv{4,6}` here, as the callers of the `installMasqueradeRules()` already do the check.

Reply:
good point, agree. fixed